cdc/puller/frontier/frontier.go

```diff
@@ -63,33 +76,50 @@ func (s *spanFrontier) Frontier() uint64 {
 }
 
 // Forward advances the timestamp for a span.
-func (s *spanFrontier) Forward(span regionspan.ComparableSpan, ts uint64) {
-	s.insert(span, ts)
+func (s *spanFrontier) Forward(regionID uint64, span regionspan.ComparableSpan, ts uint64) {
+	// it's the fast part to detect if the region is split or merged,
+	// if not we can update the minTsHeap with use new ts directly
+	if n, ok := s.cachedRegions[regionID]; ok && n.regionID > 0 && n.end != nil {
+		if regionID == n.regionID && bytes.Equal(n.Key(), span.Start) && bytes.Equal(n.End(), span.End) {
+			s.minTsHeap.UpdateKey(n.Value(), ts)
+			s.metricResolvedRegionCachedCounter.Inc()
+			return
+		}
+	}
+	s.insert(regionID, span, ts)
 }
 
-func (s *spanFrontier) insert(span regionspan.ComparableSpan, ts uint64) {
-	seekRes := s.spanList.Seek(span.Start)
-
+func (s *spanFrontier) insert(regionID uint64, span regionspan.ComparableSpan, ts uint64) {
+	// clear the  seek result
+	for i := 0; i < len(s.seekTempResult); i++ {
+		s.seekTempResult[i] = nil
+	}
+	seekRes := s.spanList.Seek(span.Start, s.seekTempResult)
 	// if there is no change in the region span
 	// We just need to update the ts corresponding to the span in list
 	next := seekRes.Node().Next()
 	if next != nil {
-		cmpStart := bytes.Compare(seekRes.Node().Key(), span.Start)
-		cmpEnd := bytes.Compare(next.Key(), span.End)
-		if cmpStart == 0 && cmpEnd == 0 {
+		if bytes.Equal(seekRes.Node().Key(), span.Start) && bytes.Equal(next.Key(), span.End) {
 			s.minTsHeap.UpdateKey(seekRes.Node().Value(), ts)
```

Comment:
Dose this line implement the same logic with the line 82~86 ?

Reply:
yes, if region is not merged or split, just update the resolved ts